File: GRASP/expert_creator/java/Venda.java
import java.time.*;
import java.util.ArrayList;

public class Venda{
    private LocalTime hora;
    private LocalDate data;
    private ArrayList<ItemVenda> itensVenda;

    public Venda(){ 
        this.data = java.time.LocalDate.now();
        this.hora = java.time.LocalTime.now();
        this.itensVenda = new ArrayList<>();
    }

    public void criarItemVenda(int quantidade, Produto produto){
        this.itensVenda.add(new ItemVenda(quantidade, produto));
    }

    public float total(){
        float total = 0;
        for (ItemVenda item: this.itensVenda){
            total += item.subtotal();
        }
        return total;
    }

    public String dataHora(){
        return this.data + " às " + this.hora;
    }
}

File: GRASP/low_coupling_high_cohesion_menu/venda.py
from pagamento import Pagamento
from datetime import date, datetime
from item_venda import ItemVenda


class Venda():
    def __init__(self):
        self._data = date.today().strftime("%d/%m/%Y")
        self._hora = datetime.now().strftime("%H:%M:%S")
        self._itens_venda = []

    def criar_item_venda(self, qtd, produto):
        self._itens_venda.append(ItemVenda(qtd, produto))

    def total(self):
        total = 0
        for item in self._itens_venda:
            total += item.subtotal()
        return total

    def data_hora(self):
        return f"{self._data} às {self._hora}"

    def pagar(self, valor_recebido):
        pagamento = Pagamento(self.total())
        return pagamento.registrar_pagamento(valor_recebido)
